wren-main/src/main/java/io/wren/main/wireprotocol/message/DescribePortalAndExecute.java

```diff
@@ -0,0 +1,102 @@
+
+import static io.wren.main.wireprotocol.message.MessageUtils.isIgnoredCommand;
+import static io.wren.main.wireprotocol.message.MessageUtils.sendHardWiredSessionProperty;
+
+public class DescribePortalAndExecute
+        implements Plan
+{
+    private static final Logger LOG = Logger.get(DescribePortalAndExecute.class.getName());
+    private final String portalName;
+    private final int maxRows;
+    private ResultSetSender resultSetSender;
+
+    public DescribePortalAndExecute(Execute parent)
+    {
+        this.portalName = parent.getPortalName();
+        this.maxRows = parent.getMaxRows();
+    }
+
+    public Optional<ResultSetSender> getResultSetSender()
+    {
+        return Optional.ofNullable(resultSetSender);
+    }
+
+    public String getPortalName()
+    {
+        return portalName;
+    }
+
+    @Override
+    public Runnable execute(Channel channel, WireProtocolSession session)
+    {
+        return () -> {
+            try {
+                Portal portal = session.getPortal(portalName);
+                String statement = portal.getPreparedStatement().getOriginalStatement();
+                if (statement.isEmpty()) {
+                    ResponseMessages.sendNoData(channel);
+                    ResponseMessages.sendEmptyQueryResponse(channel);
+                    return;
+                }
+                if (isIgnoredCommand(statement)) {
+                    ResponseMessages.sendNoData(channel);
+                    sendHardWiredSessionProperty(channel, statement);
+                    ResponseMessages.sendCommandComplete(channel, statement, 0);
+                    return;
+                }
+
+                if (!portal.isSuspended()) {
+                    Optional<ConnectorRecordIterator> connectorRecordIterable = session.execute(portalName).join();
+                    if (connectorRecordIterable.isEmpty()) {
+                        ResponseMessages.sendNoData(channel);
+                        sendHardWiredSessionProperty(channel, statement);
+                        ResponseMessages.sendCommandComplete(channel, statement, 0);
+                        return;
+                    }
+                    portal.setConnectorRecordIterator(connectorRecordIterable.get());
+                }
+
+                ConnectorRecordIterator connectorRecordIterable = portal.getConnectorRecordIterator();
+                FormatCodes.FormatCode[] resultFormatCodes = session.getResultFormatCodes(portalName);
+                resultSetSender = new ResultSetSender(
+                        statement,
+                        channel,
+                        connectorRecordIterable,
+                        maxRows,
+                        portal.getRowCount(),
+                        resultFormatCodes);
+                FormatCodes.FormatCode[] formatCodes = session.getResultFormatCodes(portalName);
+                ResponseMessages.sendRowDescription(channel, resultSetSender.getColumns(), formatCodes);
```

Comment:
We change Execute to DescribePortalAndExecute. Should it not send the result set after sending the row description?

Reply:
The reason is same as https://github.com/Canner/wren-engine/pull/509#discussion_r1544215530